src/matrix_utilities.jl

```diff
@@ -38,3 +87,44 @@ function schur_red(Π, C, R)
 end
 
 schur_red(Π, C) = schur_red(Π, C, 0.0 * I) # might be done smarter?
+
+schur_red(Π, C, R::Cholesky) = schur_red(Π, C, Matrix(R))
```

Comment:
You sure about this? It seems that you want to access `lsqrt(R)`, so keeping it a Cholesky seems useful

Reply:
Ah my bad, that's only the case when \Pi is a Cholesky too

Reply:
My current thinking is to let `\Pi` (think covariance type of filtering distribution) dictate the return type.